modPrecompiled/src/org/aion/precompiled/ContractFactory.java

```diff
@@ -78,30 +79,60 @@ public IPrecompiledContract getPrecompiledContract(
             ExecutionContext context,
             IRepositoryCache<AccountState, IDataWord, IBlockStoreBase<?, ?>> track) {
 
-        switch (context.address().toString()) {
-            case ADDR_TOKEN_BRIDGE:
-                TokenBridgeContract contract =
-                        new TokenBridgeContract(
-                                context,
-                                track,
-                                Address.wrap(ADDR_TOKEN_BRIDGE_INITIAL_OWNER),
-                                Address.wrap(ADDR_TOKEN_BRIDGE));
-
-                if (!context.origin().equals(Address.wrap(ADDR_TOKEN_BRIDGE_INITIAL_OWNER))
-                        && !contract.isInitialized()) {
+        CfgFork cfg = new CfgFork();
+        String forkProperty = cfg.getForkPropertyByNumber(1);
+        boolean firstFork =
+                (forkProperty != null) && (context.blockNumber() > Long.valueOf(forkProperty));
+
+        if (firstFork) {
+            switch (context.address().toString()) {
+                case ADDR_TOKEN_BRIDGE:
+                    TokenBridgeContract contract =
+                            new TokenBridgeContract(
+                                    context,
+                                    track,
+                                    Address.wrap(ADDR_TOKEN_BRIDGE_INITIAL_OWNER),
+                                    Address.wrap(ADDR_TOKEN_BRIDGE));
+
+                    if (!context.origin().equals(Address.wrap(ADDR_TOKEN_BRIDGE_INITIAL_OWNER))
+                            && !contract.isInitialized()) {
+                        return null;
+                    }
+
+                    return contract;
+                case ADDR_ED_VERIFY:
+                    return PC_ED_VERIFY;
+                case ADDR_BLAKE2B_HASH:
+                    return PC_BLAKE2B_HASH;
+                case ADDR_TX_HASH:
+                    return new TXHashContract(context);
+                case ADDR_TOTAL_CURRENCY:
+                default:
                     return null;
-                }
-
-                return contract;
-            case ADDR_ED_VERIFY:
-                return PC_ED_VERIFY;
-            case ADDR_BLAKE2B_HASH:
-                return PC_BLAKE2B_HASH;
-            case ADDR_TX_HASH:
-                return new TXHashContract(context);
-            case ADDR_TOTAL_CURRENCY:
-            default:
-                return null;
+            }
+        } else {
+            switch (context.address().toString()) {
+                case ADDR_TOTAL_CURRENCY:
+                    // return new TotalCurrencyContract(track, context.sender(),
+                    // Address.wrap(OWNER));
+                    return null;
```

Comment:
Suggest wee keep the code before the fork the activated. If somebody calls it after the release and before the fork number, we will be in trouble.